week-2-wide-perspective/w02e01-functional-warm-up/src/getDataFunctions.js

```diff
@@ -0,0 +1,33 @@
+export function getItemByKey(obj, key) {
+  return obj[key];
+}
+
+export function getAllItemsByKey(array, key) {
+  return array.map(item => item[key]);
+}
+
+export function getValuesStartsWith(array, key, value) {
+  return array.filter(item => item[key].toLowerCase().startsWith(value)).map(item => item[key]);
+}
+
+export function getItemsCount(array) {
+  return array.length;
+}
+
+export function getItemsSum(array, key) {
+  return array.reduce((acc, item) => acc + parseInt(item[key]), 0);
+}
+
+export function getMinValue(array, key) {
```

Comment:
ta funkcja ma za dużo "odpowiedzialności"

Jeśli coś jest `string` to sortuje po długości a jak nie to po wartości.

Albo zmiana nazwy - albo wydzielenie odpowiedzialności.
Może jesteś zbyt oszczędny w funkcje :)

Tutaj fajnie by było rzucić jakiś refactor

Reply:
Ok wydzieliłem odpowiedzialność do osobnych metod.